src/debugger.ts

```diff
@@ -14,231 +14,231 @@ const DYLD_ENV_VAR_NAME = "DYLD_INSERT_LIBRARIES";
-				.withMessage("mirrord running targetless")
-				.withDisableAction("promptTargetless")
-				.info();
-		}
-	}
-
-	if (config.type === "go") {
-		config.env["MIRRORD_SKIP_PROCESSES"] = "dlv;debugserver;compile;go;asm;cgo;link;git;gcc;as;ld;collect2;cc1";
-	} else if (config.type === "python") {
-		config.env["MIRRORD_DETECT_DEBUGGER_PORT"] = "debugpy";
-	} else if (config.type === "java") {
-		config.env["MIRRORD_DETECT_DEBUGGER_PORT"] = "javaagent";
-	}
-
-	// Add a fixed range of ports that VS Code uses for debugging.
-	// TODO: find a way to use MIRRORD_DETECT_DEBUGGER_PORT for other debuggers.
-	config.env["MIRRORD_IGNORE_DEBUGGER_PORTS"] = "45000-65535";
-
-	let isMac = platform() === "darwin";
-
-	let [executableFieldName, executable] = isMac ? getFieldAndExecutable(config) : [null, null];
-
-	let executionInfo;
-	try {
-		executionInfo = await mirrordApi.binaryExecute(target, configPath?.path || null, executable);
-	} catch (err) {
-		mirrordFailure(`mirrord preparation failed: ${err}`);
-		return null;
-	}
-
-	if (isMac) {
-		changeConfigForSip(config, executableFieldName as string, executionInfo);
-	}
-
-	let env = executionInfo?.env;
-	config.env = Object.assign({}, config.env, Object.fromEntries(env));
-
-	config.env["__MIRRORD_EXT_INJECTED"] = 'true';
-
-	return config;
+  folder: vscode.WorkspaceFolder | undefined,
+  config: vscode.DebugConfiguration,
+  _token: vscode.CancellationToken): Promise<vscode.DebugConfiguration | null | undefined> {
+  if (!globalContext.workspaceState.get('enabled')) {
+    return config;
+  }
+
+  if (config.request === "attach") {
+    new NotificationBuilder()
+      .withMessage("mirrord cannot be used with `attach` launch configurations")
+      .error();
+    return null;
+  }
+
+  // For some reason resolveDebugConfiguration runs twice for Node projects. __parentId is populated.
+  if (config.__parentId || config.env?.["__MIRRORD_EXT_INJECTED"] === 'true') {
+    return config;
+  }
+
+  updateTelemetries();
+
+  //TODO: add progress bar maybe ?
+  let cliPath;
+
+  try {
+    cliPath = await getMirrordBinary();
+  } catch (err) {
+    // Get last active, that should work?
+    cliPath = await getLastActiveMirrordPath();
+
+    // Well try any mirrord we can try :\
+    if (!cliPath) {
+      cliPath = await getLocalMirrordBinary();
+      if (!cliPath) {
+        mirrordFailure(`Couldn't download mirrord binaries or find local one in path ${err}.`);
+        return null;
+      }
+    }
+  }
+  setLastActiveMirrordPath(cliPath);
+
+  let mirrordApi = new MirrordAPI(cliPath);
+
+  config.env ||= {};
+  let target = null;
+
+  let configPath = await MirrordConfigManager.getInstance().resolveMirrordConfig(folder, config);
+  const verifiedConfig = await mirrordApi.verifyConfig(configPath, config.env);
+
+  // If target wasn't specified in the config file (or there's no config file), let user choose pod from dropdown
+  if (!configPath || (verifiedConfig && !isTargetSet(verifiedConfig))) {
```

Comment:
When I set `"target": "targetless"` in the mirrord config, the target quickpick still pops up :(

Reply:
Both `target: targetless` and not specifying a target return the same thing from config:

```json
{
  "type": "Success",
  "config": {
    "path": null,
    "namespace": null
  },
  "warnings": []
}
```

We need to change it to be explicit I guess, something like:

```json
{
  "type": "Success",
  "config": {
    "path": "targetless",
    "namespace": null
  },
  "warnings": []
}
```

Reply:
Created an issue for this in mirrord [#2010](https://github.com/metalbear-co/mirrord/issues/2010), as I'm not sure we have a good of solving this here?

Reply:
Yeah, that should probably be resolved in the CLI